src/objc-renderer.ts

```diff
@@ -243,14 +243,77 @@ function enumerationIsPublic(isPublic:boolean):(enumeration:ObjC.Enumeration) =>
   };
 }
 
+const NS_ASSUME_NONNULL_BEGIN:string = 'NS_ASSUME_NONNULL_BEGIN';
+const NS_ASSUME_NONNULL_END:string = 'NS_ASSUME_NONNULL_END';
+
+function prefixClassMacros(classInfo:ObjC.Class):string[] {
+  var macros:string[] = [];
+  switch (classInfo.nullability) {
+    case ObjC.ClassNullability.assumeNonnull:
+      macros.push(NS_ASSUME_NONNULL_BEGIN);
+      break;
+    case ObjC.ClassNullability.default:
+      break;
+  }
+  return macros;
+}
+
+function postfixClassMacros(classInfo:ObjC.Class):string[] {
+  var macros:string[] = [];
+  switch (classInfo.nullability) {
+    case ObjC.ClassNullability.assumeNonnull:
+      macros.push(NS_ASSUME_NONNULL_END);
+      break;
+    case ObjC.ClassNullability.default:
+      break;
+  }
+  return macros;
+}
+
+function prefixBlockTypeMacros(blockType:ObjC.BlockType):string[] {
+  var macros:string[] = [];
+  switch (blockType.nullability) {
+    case ObjC.ClassNullability.assumeNonnull:
+      macros.push(NS_ASSUME_NONNULL_BEGIN);
+      break;
+    case ObjC.ClassNullability.default:
+      break;
+  }
+  return macros;
+}
+
+function postfixBlockTypeMacros(blockType:ObjC.BlockType):string[] {
+  var macros:string[] = [];
+  switch (blockType.nullability) {
+    case ObjC.ClassNullability.assumeNonnull:
+      macros.push(NS_ASSUME_NONNULL_END);
```

Comment:
Something to consider is that we're also switching on this enum in four separate places. Did you think about pulling out a MacroLocations type which has four different properties (`classPrefix`, `classPostfix`, etc) or maybe even a Macro type with `prefix` and `postfix` and have MacroLocations contain two Macro types for `class` and `block`